### PR TITLE
TTAHUB-5226 - Monitoring Dashboard Finding Category filter

### PR DESCRIPTION
**Pull request by Andrew565**
## Description of change

Monitoring Dashboard Finding Category filter

## How to test

1. Load up the app, go to the regional dashboard, monitoring tab
2. Test the filter works as expected

## Jira Issue(s)

* https://jira.acf.gov/browse/TTAHUB-5226

### Every PR

- [X] Linked Jira issue
- [X] JIRA issue status updated
- [X] Code is meaningfully tested
- [X] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [X] API Documentation updated
- [X] Boundary diagram updated
- [X] Logical Data Model updated
- [X] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete
- [ ] QA review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] PR created as **Draft**
- [ ] Staging smoke test completed
- [ ] PR transitioned to **Open** _(this `ready_for_review` transition triggers the Slack/Jira automation)_
- [ ] Reviewer added after the PR is **Open** _(`seyandiangov` and `elainaparrish` are the authorized approvers under normal circumstances)_
  - _Sequence: Draft PR → Smoke test → Open PR (automation runs) → Add reviewer_
  - _Confirm that the Slack notification was sent after the PR was opened_
  - _Confirm that linked Jira ticket(s) transitioned as expected; if not, review the GitHub Actions workflow logs_

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/scopes/grantCitation/findingCategory.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Op } from 'sequelize';
+import { sequelize } from '../../models';
+
+export function withFindingCategory(categories: string[]) {
+  const escaped = categories.map((c) => sequelize.escape(c));
```
Comment (kryswisnaskas) [inline src/scopes/grantCitation/findingCategory.ts]: Codex said:

`withFindingCategory` and `withoutFindingCategory` build raw SQL from URL-derived category names after only `sequelize.escape`. This repo explicitly calls out that `escape` alone is insufficient for URL filters; those values still need independent validation before being interpolated into SQL. Finding categories are dynamic, so this cannot use a static allow-list like `findingType`, but it should still validate against active `FindingCategories` rows or use category IDs from the lookup endpoint. This keeps invalid or malformed filter input out of the SQL path and avoids oversized/odd query payloads. Add scope tests covering valid, invalid, empty, and multi-value categories.

File: frontend/src/components/filter/FilterFindingCategory.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import useFetch from '../../hooks/useFetch';
+import FilterSelect from './FilterSelect';
+import { filterSelectProps } from './props';
+
+export default function FilterFindingCategory({ onApply, inputId, query }) {
+  const { data } = useFetch(
+    [],
+    async () => {
+      return getFindingCategories();
+    },
```
Comment (thewatermethod) [inline frontend/src/components/filter/FilterFindingCategory.js]:
```suggestion
getFindingCategories,
```

File: frontend/src/components/filter/FilterFindingCategory.js
Comment (thewatermethod) [inline frontend/src/components/filter/FilterFindingCategory.js:1]: This file is missing a test

**Review by thewatermethod — APPROVED** [review]
Works well! I'd add a test for the UI and double check that everything else has a corresponding test. Also, backend tests need addressing, failures look directly related to the changes on this branch

File: src/scopes/grantCitation/findingCategory.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    where: { name: { [Op.in]: names } },
+    attributes: ['id'],
+    raw: true,
+  });
+  return rows.map((r: { id: number }) => r.id);
+}
+
+export async function withFindingCategory(categories: string[]) {
+  if (!categories.length) {
+    return { id: { [Op.in]: [] } };
```
Comment (AdamAdHocTeam) [inline src/scopes/grantCitation/findingCategory.ts:18]: Will / should this to resolve to return everything?

File: src/scopes/grantCitation/findingCategory.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const ids = await resolveValidCategoryIds(categories);
+
+  if (!ids.length) {
+    return { id: { [Op.in]: [] } };
+  }
+
+  return {
+    citationId: {
+      [Op.in]: db.sequelize.literal(
+        `(SELECT id FROM "Citations" WHERE "findingCategoryId" IN (${ids.join(',')}))`
```
Comment (AdamAdHocTeam) [inline src/scopes/grantCitation/findingCategory.ts:30]: This was flagged as possible injection on the ID's. However they come from the DB so it unlikley.

File: src/scopes/grantCitation/findingCategory.ts (hunk 3 above)
Comment (Andrew565) [reply src/scopes/grantCitation/findingCategory.ts:18]: If I take this out it returns the wrong things.

**Review by kryswisnaskas — APPROVED** [review]

**Review by kryswisnaskas — APPROVED** [review]

**Comment by Andrew565** [comment]
Need to condense filter pills.

**Review by AdamAdHocTeam — APPROVED** [review]

**Review by AdamAdHocTeam — COMMENTED** [review]
@Andrew565 looks like you having failing tests that need to be fixed before merging.
